### PR TITLE
Adding the converted Ship Game sample

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
Adding the converted Ship Game sample, direct conversion from the XNA sample.

Note, still a desktop/console focused game. 

## TODO
-  Need support for landscape on mobile
- Touch / game pad testing on mobile

### CODE REVIEW

**Review by dellis1972 — COMMENTED** [review]
Do the .config folders really need to be duplicated? can they just be at the root of the whole project?

**Comment by SimonDarksideJ** [comment]
Never tried it that way, but I can give it a go. Works make the multiplatform template a lot cleaner.
But wouldn't that affect the projects being opened individually?

**Comment by SimonDarksideJ** [comment]
As confirmed in testing @dellis1972 , having a single .config folder for the solution creates issues if you try to open just the projects themselves, and can cause the MGCB tool not to be usable for individual projects.

Unless your testing shows something different?

**Comment by dellis1972** [comment]
> As confirmed in testing @dellis1972 , having a single .config folder for the solution creates issues if you try to open just the projects themselves, and can cause the MGCB tool not to be usable for individual projects.
> 
> Unless your testing shows something different?

From the docs 

"If the file doesn't contain an entry for "isRoot": true, then also check further up the file hierarchy for additional tool manifest files.".

 So if you remove the `isRoot`, it will pick up files from above the directory.

File: ShipGame/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        },
+        {
+            "name": "ShipGame DesktopGL Debug",
+            "type": "dotnet",
+            "request": "launch",
+            "projectPath": "${workspaceFolder}/ShipGame.DesktopGL/ShipGame.DesktopGL.csproj"
+        },
+        {
+            "name": "ShipGame iOS Debug",
+            "type": "dotnet",
```
Comment (dellis1972) [inline ShipGame/.vscode/launch.json:21]: does this even work?

File: ShipGame/.vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "ShipGame Android Debug",
+            "type": "dotnet",
```
Comment (dellis1972) [inline ShipGame/.vscode/launch.json:9]: This wont work out of the box.

Comment (SimonDarksideJ) [reply ShipGame/.vscode/launch.json:9]: I just built using the dotnet tools and deployed the built apk to Android, worked fine

File: ShipGame/.vscode/launch.json (hunk 1 above)
Comment (CartBlanche) [reply ShipGame/.vscode/launch.json:21]: @dellis1972 It builds and `ShipGame.iOS.dll` is created on my Windows machine.

**Review by CartBlanche — APPROVED** [review]
Other than VSCode not running it out of the box, I think this may be ready.

**Comment by SimonDarksideJ** [comment]
> > As confirmed in testing @dellis1972 , having a single .config folder for the solution creates issues if you try to open just the projects themselves, and can cause the MGCB tool not to be usable for individual projects.
> > Unless your testing shows something different?
> 
> From the docs
> 
> "If the file doesn't contain an entry for "isRoot": true, then also check further up the file hierarchy for additional tool manifest files.".
> 
> So if you remove the `isRoot`, it will pick up files from above the directory.

OK, so I tried a root config without the "isRoot" and nothing would build.  But I tested again with the single .config at the root of the project and everything does indeed work.  WIll push that
Projects HAVE to be built from the root folder anyway as you need the .sln file for the dependencies

**Pull request merged**